src/mark.js

```diff
@@ -54,25 +54,22 @@ export class Mark {
 
 // TODO Type coercion?
 function Channel(data, {scale, type, value}) {
-  return {
-    scale,
-    type,
-    value: valueof(data, value),
-    label: value ? value.label : undefined
-  };
+  const label = value ? value.label : undefined;
+  value = value._column !== undefined && isColumnar(data) ? column(data, value._column) : valueof(data, value);
+  return {scale, type, value, label};
 }
 
 // This allows transforms to behave equivalently to channels.
 export function valueof(data, value, type) {
   const array = type === undefined ? Array : type;
-  return typeof value === "string" ? array.from(data, field(value))
+  return typeof value === "string" ? isColumnar(data) ? column(data, value) : array.from(data, field(value))
     : typeof value === "function" ? array.from(data, value)
     : typeof value === "number" || value instanceof Date ? array.from(data, constant(value))
     : value && typeof value.transform === "function" ? arrayify(value.transform(data), type)
     : arrayify(value, type); // preserve undefined type
 }
 
-export const field = label => Object.assign(d => d[label], {label});
+export const field = label => Object.assign(d => d[label], {label, _column: label});
```

Comment:
What’s up with this `_column` property? I don’t think we want to introduce a hidden property.

Reply:
The unit tests checked that channel.value was a function, so I wanted to respect that, and used this hidden property to note that it also could be a "data.column". But I'm cool with keeping channel.value as a string, it makes everything much simpler. It also breaks 61 unit tests ;-)

Reply:
I think this is the right call. Will fix the tests later.